src/exporter/DaemonMetricCollector.cc

```diff
@@ -0,0 +1,201 @@
+#include "DaemonMetricCollector.h"
+#include "common/admin_socket_client.h"
+#include "common/perf_counters.h"
+#include "include/common_fwd.h"
+
+#include <boost/json/src.hpp>
+#include <filesystem>
+#include <iostream>
+#include <regex>
+#include <string>
+#include <utility>
+
+using json_object = boost::json::object;
+using json_value = boost::json::value;
+using json_array = boost::json::array;
+
+const char *DaemonMetricCollector::SOCKETDIR = "/var/run/ceph/";
```

Comment:
On a host, cephadm should deploy an exporter per cluster since containers mount their respective cluster asok files. 

As far as I know, cephadm has `/var/run/ceph/{fsid}` folders for eachs clusters. If you create a container on a cluster it will mount `/var/run/ceph/{fsid}` to`/var/run/ceph/`, therefore, the exporter if deployed for each cluster, should parse sockets related to the cluster he is in.

Reply:
Anyways, as we plan on creating a diff PR onto the feature branch with configuration files, we can add an option to change the asok folder location.

Reply:
> Anyways, as we plan on creating a diff PR onto the feature branch with configuration files, we can add an option to change the asok folder location.

sounds good, thanks. we should try to support the non-containerized mode too, by filtering filenames in "$run_dir/$cluster-$name.asok" form by cluster prefix